services/identity/api.go

```diff
@@ -0,0 +1,226 @@
+package identity
+
+import (
+	"errors"
+	"io"
+
+	"io/ioutil"
+
+	"github.com/dedis/cothority/crypto"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/cothority/services/skipchain"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+)
+
+/*
+ */
+
+func init() {
+	for _, s := range []interface{}{
+		&Device{},
+		&Identity{},
+		&Config{},
+		&AddIdentity{},
+		&AddIdentityReply{},
+		&PropagateIdentity{},
+		&ProposeSend{},
+		&AttachToIdentity{},
+		&ProposeFetch{},
+		&ConfigUpdate{},
+		&UpdateSkipBlock{},
+		&ProposeVote{},
+	} {
+		network.RegisterMessageType(s)
+	}
+}
+
+// Identity can both follow and update an IdRoster
```

Comment:
I know you already wrote quite a lot on the subject but maybe something a bit more detailled would be nice. Either take some paragraph you already wrote, or point a link to a nicer explanation, or both.